packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.tsx

```diff
@@ -121,15 +125,9 @@ export class SpinButton extends React.Component<ISpinButtonProps, ISpinButtonSta
   // tslint:disable-next-line function-name
   public UNSAFE_componentWillReceiveProps(newProps: ISpinButtonProps): void {
     this._lastValidValue = this.state.value;
-    let value: string = newProps.value !== undefined ? newProps.value : String(newProps.min);
-    if (newProps.defaultValue) {
-      value = String(Math.max(newProps.min as number, Math.min(newProps.max as number, Number(newProps.defaultValue))));
-    }
```

Comment:
Pretty sure it is. The logic here was *super* weird... This is the line-by-line breakdown I had to go through mentally. (Keep in mind that a controlled value ought to **always** take precedence over the provided default value, or any other defaulting logic.)

```ts
let value: string = newProps.value !== undefined ? newProps.value : String(newProps.min);
```
If `newProps.value` is defined, use it (correct). Otherwise use `newProps.min` **(weird--shouldn't we just preserve the existing value?)**.

```ts
if (newProps.defaultValue) {
```
If `newProps.defaultValue` is non-falsey **(mishandles `''`)**...
```ts
  value = String(Math.max(newProps.min, Math.min(newProps.max, Number(newProps.defaultValue))));
```
...unconditionally use it **(possibly overriding `newProps.value`--wrong)**, clamping it within min/max range **(weird since the constructor doesn't do this clamping with `defaultValue`)**. (Also missing a check for `isNaN(Number(defaultValue))` which is needed since `value`/`defaultValue` can have non-numeric text too.)

```ts
if (newProps.value !== undefined) {
```
Only if `newProps.value` is defined (which would be right, but is **weird because we might have just overridden it based on `defaultValue`...?**)
```ts
  this.setState({ value: value });
```
Update state.

So the only case where this is a behavior change is if someone passes in both `value` and `defaultValue` and then updates props, but that's wrong to begin with and we have a `warnMutuallyExclusive` for it in the constructor, so hopefully it's not too likely.